core/domain/stats_services.py

```diff
@@ -16,251 +16,137 @@
 
     Args:
-        exploration_id: str. The exploration ID.
-        state_name: str. Name of the state.
-
-    Returns:
-        list(dict). Each item in the list is a dict with keys representing
-        - 'id': str. The visualization ID.
-        - 'data': list(dict). A list of calculation IDs.
-        - 'options': dict. The visualization options.
-
-        An example of the returned value may be:
-        [{'options': {'y_axis_label': 'Count', 'x_axis_label': 'Answer'},
-        'id': 'BarChart',
-        'data': [{u'frequency': 1, u'answer': 0}]}]
+        exploration: Exploration. The exploration domain object.
     """
-    exploration = exp_services.get_exploration_by_id(exploration_id)
-    if exploration.states[state_name].interaction.id is None:
-        return []
-
-    visualizations = interaction_registry.Registry.get_interaction_by_id(
-        exploration.states[state_name].interaction.id).answer_visualizations
-
-    calculation_ids = set([
-        visualization.calculation_id for visualization in visualizations])
+    state_stats_mapping = {}
+    for state_name in exploration.states:
+        state_stats_mapping[state_name] = (
+            stats_domain.StateStats.create_default())
 
-    calculation_ids_to_outputs = {}
-    for calculation_id in calculation_ids:
-        # This is None if the calculation job has not yet been run for this
-        # state.
-        calc_output_domain_object = (
-            stats_jobs_continuous.InteractionAnswerSummariesAggregator.get_calc_output( # pylint: disable=line-too-long
-                exploration_id, state_name, calculation_id))
+    exploration_stats = stats_domain.ExplorationStats(
+        exploration.id, exploration.version, 0, 0, state_stats_mapping)
+    create_stats_model(exploration_stats)
 
-        # If the calculation job has not yet been run for this state, we simply
-        # exclude the corresponding visualization results.
-        if calc_output_domain_object is None:
-            continue
 
-        calculation_ids_to_outputs[calculation_id] = (
-            calc_output_domain_object.calculation_output)
-    return [{
-        'id': visualization.id,
-        'data': calculation_ids_to_outputs[visualization.calculation_id],
-        'options': visualization.options,
-    } for visualization in visualizations
-            if visualization.calculation_id in calculation_ids_to_outputs]
-
-
-# TODO(bhenning): Test
-def get_versions_for_exploration_stats(exploration_id):
-    """Returns a list of strings, each string representing a version of the
-    given exploration_id for which statistics data exists. These versions are
-    retrieved from ExplorationAnnotationsModel created when StatisticsAggregator
-    job is run.
-
-    An example of the return list may look like [u'3', u'all']
-    where '3' and 'all' are versions of the given exploration ID from
-    ExplorationAnnotationsModel.
+def handle_stats_creation(exploration, change_list):
```

Comment:
handle_stats_creation_for_new_exp_version, maybe

But, similarly to the above, I'm not sure why this is even a deferred thing in the first place.

Reply:
Re the naming of these two functions, I've changed it to something better. PTAL